### PR TITLE
[PaddleDetection] add save best model

### PR DESCRIPTION
**Pull request by yghstill**
1. add save best model function

-----Local Test------
image
image

2. fix no bbox in eval crash
before fix:
image
after fix:
image

3. fix dcn configs

### CODE REVIEW

**Review by jerrywgz — COMMENTED** [review]
Need related doc for best model in --eval

**Comment by yghstill** [comment]
> Need related doc for best model in --eval

DONE

File: PaddleCV/PaddleDetection/docs/GETTING_STARTED.md [hunk 1]
```diff
@@ -51,7 +51,8 @@ python -u tools/train.py -c configs/faster_rcnn_r50_1x.yml --eval
 
 Alternating between training epoch and evaluation run is possible, simply pass
 in `--eval` to do so and evaluate at each snapshot_iter. It can be modified at `snapshot_iter` of the configuration file. If evaluation dataset is large and
-causes time-consuming in training, we suggest decreasing evaluation times or evaluating after training.
+causes time-consuming in training, we suggest decreasing evaluation times or evaluating after training. When perform evaluation in training, save best model(mAP best)
```
Comment (jerrywgz) [inline PaddleCV/PaddleDetection/docs/GETTING_STARTED.md]: the best model with highest MAP is saved at each `snapshot_iter` 
另外说明下best model的路径和model_final相同吧

**Review by jerrywgz — APPROVED** [review]
LGTM

**Pull request merged**